### PR TITLE
Fixed issue #2118

### PR DESCRIPTION
**Pull request by pieterb**
## Are there changes in behavior for the user?

The signature of `DynamicResource.__init__()` changed. This signature is not (yet) part of the public interface, and probably not used by third party developers, as its mostly useless in its old form.

## Related issue number

#2118 

## Checklist

- [*] I think the code is well written
- [*] Unit tests for the changes exist
- [n/a] Documentation reflects the changes
- [*] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [*] Add a new news fragment into the `changes` folder
  * name it `<issue_id>.<type>` for example (588.bug)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

File: changes/2118.feature [hunk 1]
```diff
@@ -0,0 +1 @@
+Refactor DynamicResource construction to facilitate subclassing from DynamicResource.
```
Comment (asvetlov) [inline changes/2118.feature]: Please drop the file: the change is not related to public API, I don't want to attract attention by this record

Comment (pieterb) [reply changes/2118.feature]: Done!
